sys/include/irq.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+* Copyright (C) 2021 Jens Wetterich <jens@wetterich-net.de>
+*
+* This file is subject to the terms and conditions of the GNU Lesser
+* General Public License v2.1. See the file LICENSE in the top level
+* directory for more details.
+*/
+/**
+* @ingroup core_irq
+* @{
+* @file
+* @brief   Provides a C++ RAI based API to control interrupt processing
+* @author Jens Wetterich <jens@wetterich-net.de>
+*
+*/
+#ifndef IRQ_HPP
+#define IRQ_HPP
+#include "irq.h"
+
+namespace riot {
+/**
+ * @brief RAII based IRQ lock
+ * @details
+ * While this object is on the stack IRQ is disabled.
+ * During destruction it will be restored to the previous state.
```

Comment:
```suggestion
 * @brief   RAII based IRQ lock
 * @details While this object is on the stack IRQ is disabled.
 *          During destruction it will be restored to the previous state.
```